### PR TITLE
fix: enabling npm artifacts repo process update (rhdevdocs-5035)

### PR DESCRIPTION
**Pull request by deerskindoll**
## What does this pull request change?
enabling npm artifacts  process update

## What issues does this pull request fix or reference?
https://issues.redhat.com/browse/RHDEVDOCS-5035

## Specify the version of the product this pull request applies to
main

## Pull Request checklist

The author and the reviewers validate the content of this pull request with the following checklist, in addition to the [automated tests](code_review_checklist.adoc).

- Any procedure:
  - [ ] Successfully tested.
- Any page or link rename:
  - [ ] The page contains a redirection for the previous URL.
  - Propagate the URL change in:
    - [ ] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/packages/dashboard-frontend/src/services/bootstrap/branding.constant.ts)
- [ ] Builds on [Eclipse Che hosted by Red Hat](https://workspaces.openshift.com).
- [ ] the *`Validate language on files added or modified`* step reports no vale warnings.

### CODE REVIEW

File: modules/end-user-guide/partials/proc_restoring-node-extra-ca-certs-variable-value.adoc [hunk 1]
```diff
@@ -0,0 +1,14 @@
+[id="restoring-node-extra-ca-certs-variable-value"]
+= Restoring correct NODE_EXTRA_CA_CERTS variable value
+
+Use the command described below to restore the correct NODE_EXTRA_CA_CERTS variable value every time before dependencies are downloaded.
```
Comment (nickboldt) [inline modules/end-user-guide/partials/proc_restoring-node-extra-ca-certs-variable-value.adoc]: Seems a big chonky.

```suggestion
Use the command below to set NODE_EXTRA_CA_CERTS to point to where you have your SSL/TLS certificate. 
```

File: modules/end-user-guide/partials/proc_restoring-node-extra-ca-certs-variable-value.adoc [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+.Procedure
+
+* Run the following command in the terminal:
++
+[source, yaml]
+----
+`export NODE_EXTRA_CA_CERTS=/public-certs/nexus.cer` <1>
```
Comment (nickboldt) [inline modules/end-user-guide/partials/proc_restoring-node-extra-ca-certs-variable-value.adoc:12]: per https://stackoverflow.com/questions/23788564/npm-add-root-ca/56239954#56239954

```suggestion
`export NODE_EXTRA_CA_CERTS=/public-certs/nexus.cer` <1>

Then download dependencies with:
`npm install` or `yarn`
```

File: modules/end-user-guide/partials/proc_disabling-self-signed-certificate-validation.adoc [hunk 3]
```diff
@@ -0,0 +1,13 @@
+[id="disabling-self-signed-certificate-validation"]
+= Disabling self-signed certificate validation
+
+Follow the instructions below to disable self-signed certificate validation.
```
Comment (nickboldt) [inline modules/end-user-guide/partials/proc_disabling-self-signed-certificate-validation.adoc]:
```suggestion
Run the command below to disable SSL, which will bypass validation of your self-signed certificates. Note that this is a potential security risk. For a better solution, configure use of a self-signed certificate you trust with `NODE_EXTRA_CA_CERTS`. 
```

File: modules/end-user-guide/partials/proc_restoring-node-extra-ca-certs-variable-value.adoc [hunk 4]
```diff
@@ -0,0 +1,14 @@
+[id="restoring-node-extra-ca-certs-variable-value"]
+= Restoring correct NODE_EXTRA_CA_CERTS variable value
```
Comment (nickboldt) [inline modules/end-user-guide/partials/proc_restoring-node-extra-ca-certs-variable-value.adoc]:
```suggestion
= Configuring NODE_EXTRA_CA_CERTS to use a certificate
```

File: modules/end-user-guide/partials/proc_restoring-node-extra-ca-certs-variable-value.adoc [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+Use the command described below to restore the correct NODE_EXTRA_CA_CERTS variable value every time before dependencies are downloaded.
+
+.Procedure
+
+* Run the following command in the terminal:
++
+[source, yaml]
```
Comment (nickboldt) [inline modules/end-user-guide/partials/proc_restoring-node-extra-ca-certs-variable-value.adoc]:
```suggestion
[source, bash]
```

**Comment by tolusha** [comment]
```
kind: Secret
apiVersion: v1
metadata:
  name: tls-cer
  annotations:
    controller.devfile.io/mount-path: /public-certs/nexus.cer
    controller.devfile.io/mount-as: file
  labels:
    controller.devfile.io/mount-to-devworkspace: 'true'
    controller.devfile.io/watch-secret: 'true'
data:
  tls.cer: >-
    <Base64_encoded_content_of_public_cert>
```

**Comment by tolusha** [comment]
The correct Secret should look like:

```
kind: Secret
apiVersion: v1
metadata:
  name: tls-cer
  annotations:
    controller.devfile.io/mount-path: /public-certs
    controller.devfile.io/mount-as: file
  labels:
    controller.devfile.io/mount-to-devworkspace: 'true'
    controller.devfile.io/watch-secret: 'true'
data:
  nexus.cer: >-
    <Base64_encoded_content_of_public_cert>
```

You can find the template for tests in disconnected environment: https://gitlab.cee.redhat.com/codeready-workspaces/crw-jenkins/-/blob/master/resources/disconnected/nexus/tls-nexus-crt.secret.yaml.template

**Comment by deerskindoll** [comment]
> The correct Secret should look like:
> 
> ```
> kind: Secret
> apiVersion: v1
> metadata:
>   name: tls-cer
>   annotations:
>     controller.devfile.io/mount-path: /public-certs
>     controller.devfile.io/mount-as: file
>   labels:
>     controller.devfile.io/mount-to-devworkspace: 'true'
>     controller.devfile.io/watch-secret: 'true'
> data:
>   nexus.cer: >-
>     <Base64_encoded_content_of_public_cert>
> ```
> 
> You can find the template for tests in disconnected environment: [gitlab.cee.redhat.com/codeready-workspaces/crw-jenkins/-/blob/master/resources/disconnected/nexus/tls-nexus-crt.secret.yaml.template](https://gitlab.cee.redhat.com/codeready-workspaces/crw-jenkins/-/blob/master/resources/disconnected/nexus/tls-nexus-crt.secret.yaml.template)

I checked the preview of the whole page and the secret is described there: https://6531a534185e4f0ce74a6fec--eclipse-che-docs-pr.netlify.app/docs/next/end-user-guide/enabling-npm-artifact-repositories/. UPDATE: Now I see the difference. Let me update it.

**Review by tolusha — APPROVED** [review]

**Pull request merged**